### DISCUSSION TITLE
Thread with Smallrye Messaging

### DISCUSSION THREAD

**Discussion by antoinechamot** [quarkusio/quarkus#34113] [Q&A]
I have a message consumer like this one in Quarkus that consume messages in a queue :

```java
@Incoming("incoming-messages") 
@Blocking(ordered = false, value = "custom-worker-pool")
public Uni<Void> onExecutionReceive(final Message<JsonObject> message) { 
       message.ack();
       // Do long process here 
       return Uni.createFrom().voidItem();
}

``` 

With associated configuration in application.properties :
```
smallrye.messaging.worker.custom-worker-pool.max-concurrency=3
``` 

This works well : A vertex worker thread consume my messages. But the problem occurs when my blocking process exceed 60s i get this warning :
```shell
.. has been blocked for 61159 ms, time limit is 60000 ms: io.vertx.core.VertxException: Thread blocked
``` 

In order to avoid that i would replace the vertex worker thread used by smallrye by a regular thread pool that has no time limitation for example something like Executors.newFixedThreadPool(3).

Is it possible to do so ? If yes how ?

[a comment by an automated account is not shown]

**Comment by ozangunalp** [comment]
Using the Vert.x worker threads allow us to do context propagation.
In Quarkus you can set the global timeout for Vert.x worker executors with [quarkus.vertx.max-worker-execute-time](https://quarkus.io/guides/all-config#quarkus-vertx_quarkus.vertx.max-worker-execute-time) 

What we can do is add a config property like `smallrye.messaging.worker.custom-worker-pool.max-worker-execute-time` similar to how `max-concurrency` per custom worker pool works.

**Reply by antoinechamot** [reply]
That would be perfect. Do you want me to open a ticket for this ?

**Reply by HugoMRAmaro** [reply]
Could you please tell me if there are any news regarding this? thank you

**Reply by cescoffier** [reply]
No news - please open an issue if needed.
